### PR TITLE
Updated the session proxy article

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | #5844 |

I'm not sure if this is what @xabbuh was talking about in the related issue.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Could I get a review for these changes? Thanks!

File: cookbook/session/proxy_examples.rst [hunk 1]
```diff
@@ -24,6 +79,9 @@ Encryption of Session Data
 If you wanted to encrypt the session data, you could use the proxy to encrypt
 and decrypt the session as required::
 
+    // src/AppBundle/Session/EncryptedSessionProxy.php
+    namespace AppBundle/Session;
```
Comment (HeahDude) [inline cookbook/session/proxy_examples.rst]: Namespaces use backslashes `\`

File: cookbook/session/proxy_examples.rst [hunk 2]
```diff
@@ -59,10 +117,13 @@ There are some applications where a session is required for guest users, but
 where there is no particular need to persist the session. In this case you
 can intercept the session before it is written::
 
-    use Foo\User;
+    // src/AppBundle/Session/ReadOnlySessionProxy.php
+    namespace AppBundle/Session;
```
Comment (HeahDude) [inline cookbook/session/proxy_examples.rst]: same here :)

**Comment by HeahDude** [comment]
I left some very minor comments but I like your changes :+1:

File: cookbook/session/proxy_examples.rst (hunk 1 above)
Comment (javiereguiluz) [inline cookbook/session/proxy_examples.rst]: :scream: Nice catch! Fixed. Thanks!

File: cookbook/session/proxy_examples.rst [hunk 3]
```diff
@@ -4,19 +4,74 @@
...
+            </service>
+        </services>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->setDefinition('app.session_handler', new Definition('AppBundle\Session\CustomSessionHandler'));
```
Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]: you can simply use `register()`

File: cookbook/session/proxy_examples.rst [hunk 4]
```diff
@@ -4,19 +4,74 @@
...
+        services:
+            app.session_handler:
+                class: AppBundle\Session\CustomSessionHandler
+
+    .. code-block:: xml
+
+        <services>
+            <service id="app.session_handler"
+                class="AppBundle\Session\CustomSessionHandler">
+            </service>
```
Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]: Let's use this:

```xml
<service id="app.session_handler" class="AppBundle\Session\CustomSessionHandler" />

File: cookbook/session/proxy_examples.rst [hunk 5]
```diff
@@ -4,19 +4,74 @@
...
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.session_handler:
+                class: AppBundle\Session\CustomSessionHandler
+
+    .. code-block:: xml
+
+        <services>
```
Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]: missing file header comment:

```xml
<!-- app/config/services.xml -->
```

File: cookbook/session/proxy_examples.rst [hunk 6]
```diff
@@ -4,19 +4,74 @@
...
+    .. code-block:: php
+
+        // app/config/config.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->setDefinition('app.session_handler', new Definition('AppBundle\Session\CustomSessionHandler'));
+
+Finally, use the ``framework.session.handler_id`` configuration option to tell
+Symfony yo use your own session handler instead of the default one:
```
Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]: [...] to use [...]

File: cookbook/session/proxy_examples.rst [hunk 7]
```diff
@@ -4,19 +4,74 @@
...
+
+        <!-- app/config/config.xml -->
+        <framework:config>
+            <framework:session handler-id="app.session_handler" ... />
+        </framework:config>
+
+    .. code-block:: php
+
+        // app/config/config.php
+
```
Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]: this blank line should be removed

File: cookbook/session/proxy_examples.rst [hunk 8]
```diff
@@ -4,19 +4,74 @@
...
+        $container->loadFromExtension('framework', array(
+            ...,
```
Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]:
```php
// ...
```

File: cookbook/session/proxy_examples.rst (hunk 3 above)
Comment (javiereguiluz) [inline cookbook/session/proxy_examples.rst]: I'm sorry but I don't understand this comment.

**Comment by javiereguiluz** [comment]
I've fixed most of the issues reported by reviewers.

File: cookbook/session/proxy_examples.rst (hunk 3 above)
Comment (ogizanagi) [inline cookbook/session/proxy_examples.rst]: @javiereguiluz : 
```diff
- $container->setDefinition('app.session_handler', new Definition('AppBundle\Session\CustomSessionHandler'));
+ $container->register('app.session_handler', 'AppBundle\Session\CustomSessionHandler');
```

Comment (javiereguiluz) [inline cookbook/session/proxy_examples.rst]: Thanks @ogizanagi! It's fixed now.

File: cookbook/session/proxy_examples.rst [hunk 9]
```diff
@@ -4,26 +4,79 @@
...
+        services:
+            app.session_handler:
+                class: AppBundle\Session\CustomSessionHandler
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <services>
+            <service id="app.session_handler" class="AppBundle\Session\CustomSessionHandler" />
+        </services>
```
Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]: I think we should use a complete XML example here and below.

Comment (xabbuh) [inline cookbook/session/proxy_examples.rst]: So it should be:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://symfony.com/schema/dic/services
        http://symfony.com/schema/dic/services/services-1.0.xsd">

    <services>
        <service id="app.session_handler" class="AppBundle\Session\CustomSessionHandler" />
    </services>
</container>
```

**Comment by xabbuh** [comment]
@javiereguiluz Apart from the XML config examples this looks ready to me.

**Comment by javiereguiluz** [comment]
I've updated the examples to show the full XML config. Thanks!

**Comment by xabbuh** [comment]
:+1:

**Comment by xabbuh** [comment]
ping @weaverryan @WouterJ

**Comment by WouterJ** [comment]
Thanks Javier, it looks great!

**Pull request closed without merging**